napari/viewer.py

```diff
@@ -0,0 +1,27 @@
+from .components import Window, Viewer
+
+
+class ViewerApp:
+    """Napari ndarray viewer.
+
+    Parameters
+    ----------
+    *images : ndarray
+        Arrays to render as image layers.
+    **named_images : dict of str -> ndarray
+        Arrays to render as image layers, keyed by layer name.
+    """
+    def __init__(self, *images, **named_images):
```

Comment:
I like being able to pass either one image or the named images, that's very nice. Do we also want the ability to set the keyword arguments of `add_image` too, such as the meta data? As a separate PR I think we want to make things like `clims`, `colormap` etc. settable in the constructor of `add_image` and I think we want to be able to just propagate any of those things through this part of the API too

Reply:
I did think about that, I'm happy to do that, but here are the caveats:

- any named parameters will be forbidden as image names
- the same parameters will apply to all images. I actually started down the rabbit hole of passing a list or dictionary of configs, but that got real ugly real fast, and I decided against it.

Reply:
Side note, clims needs a name change

Reply:
Hmm you make good points. I agree that the dictionary approach is not too appealing, but I don't think I mind the extra keyword args applying to all image layers. I view this as convenience functionality and as such might have to be used with some caution.

It would be a shame to lose the ability to have named images completely as you do in the current examples, but it also doesn't really bother me to have a couple "protected" keywords that cannot be used for names, such as `colormap` etc.

Curious what @kne42 thinks

Reply:
I think as long as you "reserve" the keyword arguments by explicitly including them in the signature definition, it should be clear enough that it won't get too confusing.